Update iCloud integration documentation

stankiewiczmichal opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
I clarified password requirements for iCloud integration setup. Earlier it was written that app-specific passwords should be used. However, they currently [do not work](https://github.com/home-assistant/core/issues/128830#issuecomment-3572235888), so I noted that it cannot be done and that one should use their default Apple Account password instead. In addition, I removed the mention that two-factor authentication does not work because it does work, and I noted that work is underway to fix the bug that causes users to receive emails and notifications about attempts to log into their Apple Account. 
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue:
- This PR is connected with issue #128830

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

stankiewiczmichal committed:
Update iCloud integration documentation

Clarified password requirements for iCloud integration setup.

stankiewiczmichal committed:
Merge branch 'current' into patch-1

frenck commented on source/_integrations/icloud.markdown:
```diff
@@ -28,20 +28,21 @@ It does require that your devices are registered with the [Find My](https://www.
 
 For the notification, press "Allow", then "OK".
+
+Work is currently underway to resolve this problem. 
```
This shouldn't be written, as this set expectations that might never be resolved.

stankiewiczmichal replied:
I understand, it will be corrected.

frenck commented on source/_integrations/icloud.markdown:
```diff
@@ -28,20 +28,21 @@ It does require that your devices are registered with the [Find My](https://www.
 
 {% include integrations/config_flow.md %}
 
+{% important %}
+You can't use an [app-specific password](https://support.apple.com/102654). You need to use your Apple Account password to set up this integration.
+{% endimportant %}
```
Should this be a prerequisite?

stankiewiczmichal replied:
In the current version of the extension, unfortunately, there is such a requirement, and nothing can be done about it.

frenck replied:
I disagree, this same thing can be re-written into a requisite. Instead of telling what doesn't work, explain what does work and is needed to succeed.

stankiewiczmichal committed:
Merge branch 'current' into patch-1

stankiewiczmichal committed:
Clean up iCloud integration warnings in documentation

Removed unnecessary warnings about account login notifications and ongoing work.

stankiewiczmichal committed:
Merge branch 'current' into patch-1
